utils/spid_blob_storage.ts

```diff
@@ -0,0 +1,85 @@
+import { BlobService } from "azure-storage";
+import * as azureStorage from "azure-storage";
+import { Either, either, left, right } from "fp-ts/lib/Either";
+
+export function appendSpidBlob(
+  blobService: BlobService,
+  containerName: string,
+  blobName: string,
+  spidMsgItem: string
+): Promise<Either<Error, azureStorage.BlobService.BlobResult>> {
+  return new Promise(resolve => {
+    blobService.createContainerIfNotExists(containerName, (err, _, __) => {
```

Comment:
try to use taskify to convert callback based method to promise based: 
https://github.com/gcanti/fp-ts/blob/1.17.0/src/TaskEither.ts#L334